### PR TITLE
address_module

### PR DESCRIPTION
**Pull request by nikpawar89**

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/commands/domain/CommandSource.java [hunk 1]
```diff
@@ -129,8 +130,11 @@ public void markAsRejected(final AppUser checker, final DateTime checkedOnDate){
         this.checkedOnDate = checkedOnDate.toDate();
         this.processingResult = CommandProcessingResultType.REJECTED.getValue();
     }
+    
+
+   
 
-    public void updateResourceId(final Long resourceId) {
+	public void updateResourceId(final Long resourceId) {
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/commands/domain/CommandSource.java]: This file should not be part of pull request as it is having only formatting issue

File: fineract-provider/src/main/java/org/apache/fineract/commands/domain/CommandWrapper.java [hunk 2]
```diff
@@ -38,6 +38,8 @@
     private final String json;
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/commands/domain/CommandWrapper.java]: This file should not be part of pull request as it is having only formatting issue

File: fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java [hunk 3]
```diff
@@ -41,10 +41,37 @@
...
-                this.entityName, this.entityId, this.subentityId, this.href, this.json, this.transactionId, this.productId, this.templateId);
+                this.entityName, this.entityId, this.subentityId, this.href, this.json, this.transactionId, this.productId,
+                this.templateId);
+    }
+    
+    public CommandWrapperBuilder addClientAddress(final long clientId,final long addressTypeId) {
+        this.actionName = "CREATE";
+        this.entityName = "ADDRESS";
+        this.entityId = addressTypeId;
+        this.href = "/Address/template";
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java]: href doesn't show correct url

File: fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java [hunk 4]
```diff
@@ -41,10 +41,37 @@
...
+        this.clientId=clientId;
+        return this;
+    }
+    
+    public CommandWrapperBuilder updateClientAddress(final long clientId) {
+        this.actionName = "UPDATE";
+        this.entityName = "ADDRESS";
+        this.href = "/Address/template";
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java]: href doesn't show correct url

File: fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java [hunk 5]
```diff
@@ -41,10 +41,37 @@
...
+        this.clientId=clientId;
+        return this;
+    }
+    
+    public CommandWrapperBuilder addEntityFieldConfiguration(final String entityname) {
+        this.actionName = "MAP";
+        this.entityName = entityname.toUpperCase();
+        this.entityId = null;
+        this.href = "/Address/template";
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java]: href doesn't show correct url

File: fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+-- under the License.
+--
+
+--code inserts
+INSERT INTO `m_code` (`id`, `code_name`, `is_system_defined`) VALUES (150, 'INDIAN STATES', 0);
+INSERT INTO `m_code` (`id`, `code_name`, `is_system_defined`) VALUES (151, 'COUNTRY ID', 0);
+INSERT INTO `m_code` (`id`, `code_name`, `is_system_defined`) VALUES (152, 'ADDRESS TYPE ID', 0);
+
+--add new column in portfolio
+alter table `m_portfolio_command_source` add column `status` TINYINT(4) NULL DEFAULT NULL;
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql]: with current approach you no need to alter m_portfolio_command_source table

File: fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+-- Unless required by applicable law or agreed to in writing,
+-- software distributed under the License is distributed on an
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied. See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+--
+
+--code inserts
+INSERT INTO `m_code` (`id`, `code_name`, `is_system_defined`) VALUES (150, 'INDIAN STATES', 0);
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql]: You should not insert id as they are auto incremented and the insert will fail if m_code table already has these ids

File: fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+INSERT INTO `m_code` (`id`, `code_name`, `is_system_defined`) VALUES (151, 'COUNTRY ID', 0);
+INSERT INTO `m_code` (`id`, `code_name`, `is_system_defined`) VALUES (152, 'ADDRESS TYPE ID', 0);
+
+--add new column in portfolio
+alter table `m_portfolio_command_source` add column `status` TINYINT(4) NULL DEFAULT NULL;
+
+--configuration
+INSERT INTO `c_configuration` (`id`, `name`, `value`, `date_value`, `enabled`, `is_trap_door`, `description`) VALUES ('Enable-Address', NULL, NULL, 1, 0, NULL);
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql]: This insert statement will fail as parameters and actual values length is not matching

File: fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql (hunk 7 above)
Comment (nikpawar89) [inline fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql]: would change it

File: fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql (hunk 8 above)
Comment (nikpawar89) [inline fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql]: would alter it

File: fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql (hunk 6 above)
Comment (nikpawar89) [inline fineract-provider/src/main/resources/sql/migrations/core_db/v312_address_module_tables_metadat.sql]: ok.. would not alter the table

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java [hunk 9]
```diff
@@ -76,29 +76,44 @@ public GlobalConfigurationApiResource(final PlatformSecurityContext context,
     @GET
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public String retrieveConfiguration(@Context final UriInfo uriInfo,@DefaultValue("false") @QueryParam("survey") final boolean survey) {
+    public String retrieveConfiguration(@Context final UriInfo uriInfo,
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java:79]: Why do you need to change GlobalConfigurationApiResource API definitions?

Comment (nikpawar89) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java:79]: Hi Nazeer,
This is the new configuration: 
public String retrieveConfiguration(@Context final UriInfo uriInfo,@QueryParam("configName") final String configName,@DefaultValue("false") @QueryParam("survey") final boolean survey)

I have added configName as new parameter, beacuse we are going to fetch configuration by the name

Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java:79]: I don't know whether you have followed our discussion. Let me put it here what I understood. You are supposed to add a new method in read platform service. Not in the API resource. Let me know if you differ this :-)

Comment (nikpawar89) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java]: got your point.. i will undo changes in API.. sorry for the confusion

Comment (nikpawar89) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java:79]: Ok let me explain you, why I needed to add additional parameter, configname in APi.
In Community App, we have added a functionality wherein, only if address configuration is enabled, the address of the clients would be displayed. 
In order to make this functionality work independent of id, I had to modify the api, to get the configuration by name.

**Pull request closed without merging**